cdap-docs/tools/docs-extract-table.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Used to generate a table in the CLI documentation from a table in a Markdown file.
+#
+
+from optparse import OptionParser
+import os
+import sys
+
+VERSION = "0.0.1"
+
+SPACES = ' ' * 3
+PIPE = '|'
+DOUBLE_QUOTE = '"'
+TABLE_HEADER = """.. csv-table::
+   :header: Command,Description
+   :widths: 40, 60
+
+"""
+MISSING_FILE_TEMPLATE = "   **Missing Input File**,\"Missing input file %s\""
+
+
+def parse_options():
+    """ Parses args options.
+    """
+    parser = OptionParser(
+        usage="\n\n  %prog input_file output_file",
+        description="Generates an rst table to be included in the documentation")
+
+    parser.add_option(
+        "-v", "--version",
+        action="store_true",
+        dest="version",
+        help="Version of this software",
+        default=False)
+
+    (options, args) = parser.parse_args()
+    
+    if options.version:
+        print "Version: %s" % VERSION
+        sys.exit(1)
+
+    if not args or len(args) != 2:
+        parser.print_help()
+        sys.exit(1)
+
+    return options, args
+
+#
+# Utility functions
+#
+
+def create_parsed_line(line):
+    """ Parses a line from a Markdown table and converts it to a reST CSV table line.
+    """
+
+    pieces = [piece.strip() for piece in line.split(PIPE)][1:-1]
+    pieces[0] = "``%s``" % pieces[0]
+    pieces = ["\"%s\"" % piece for piece in pieces]
+    comma = ','
+    return comma.join(pieces)
+
+def create_from_lines(line1, line2):
+    """ Combines two lines from a Markdown table and converts them into a single line for a reST CSV table line.
+    """
+    pieces1 = [piece.strip() for piece in line1.split(PIPE)][1:-1]
+    pieces2 = [piece.strip() for piece in line2.split(PIPE)][1:-1]
+    
+    if len(pieces1) != len(pieces2):
+        return ''
+        
+    pieces = []
+    for i in range(len(pieces1)):    
+        pieces.append(pieces1[i] + ' ' + pieces2[i])
+    
+    pieces = [piece.strip() for piece in pieces]
```

Comment:
Also, line 80 seems redundant, because we already strip the pieces in lines 70 and 71.